### ISSUE TITLE
Documentation for UpdateByQuery

### ISSUE THREAD

**Issue by PhaedrusTheGreek** [elastic/elasticsearch#18787]
The new UpdateByQuery is awesome!  But lacks Java API documentation.   

@debadair ,

Speaking with @nik9000 , I formulated & tested this:

```
UpdateByQueryRequestBuilder updateByQuery = UpdateByQueryAction.INSTANCE.newRequestBuilder(client);

updateByQuery.source("cool_things")
        .filter(termQuery("level", "awesome"))
        .script(new Script("ctx._source.awesome = \"absolutely\""));

BulkIndexByScrollResponse response = updateByQuery.get();
```

**Comment by PhaedrusTheGreek** [comment]
I forgot to mention a few important points

You must have the reindex module installed.  Here's the POM example for 2.3.2:

```
		<dependency>
			<groupId>org.elasticsearch.module</groupId>
			<artifactId>reindex</artifactId>
			<version>2.3.2</version>
		</dependency>
```

And you have to add the ReindexPlugin class to the client:

```
Client client = TransportClient.builder()
		.settings(settings)
		.addPlugin(org.elasticsearch.index.reindex.ReindexPlugin.class)
		.build() 
		.addTransportAddress(new InetSocketTransportAddress(new InetSocketAddress("localhost", 9300)));
```

**Comment by NicolasYDDER** [comment]
+1

**Comment by henningandersen** [comment]
@PhaedrusTheGreek since we recommend to not use the transport client and put all documentation efforts into the new high level rest API, I think this issue can be closed? See also #24604.

**Comment by henningandersen** [comment]
@PhaedrusTheGreek since the transport client and the associated java API docs are now gone in master, I am closing this issue.
